snuba_sdk/request.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import json
+import re
+from dataclasses import dataclass, fields
+
+from snuba_sdk.query import Query
+
+
+class InvalidRequestError(Exception):
+    pass
+
+
+class InvalidFlagError(Exception):
+    pass
+
+
+@dataclass(frozen=True)
+class Flags:
+    totals: bool | None = None
+    consistent: bool | None = None
+    turbo: bool | None = None
+    debug: bool | None = None
+    dry_run: bool | None = None
+    legacy: bool | None = None
```

Comment:
No I don't want them to be sent in the payload unless they are explicitly specified.

Reply:
Usually, in the Snuba codebase we write something like:
```python3
arg: Optional[type] = None
```
Any reason you're doing `bool | None` instead of `Optional[bool]`?

Reply:
It's a newer format that I think is cleaner. When we typed Snuba initially either this format wasn't supported or we didn't know about it. It has fewer imports and is a little easier to read.

https://peps.python.org/pep-0604/